graphql/graphql.go

```diff
@@ -997,16 +1014,8 @@ func (b *Block) Logs(ctx context.Context, args struct{ Filter BlockFilterCriteri
 	if args.Filter.Topics != nil {
 		topics = *args.Filter.Topics
 	}
-	hash := b.hash
-	if hash == (common.Hash{}) {
-		header, err := b.resolveHeader(ctx)
-		if err != nil {
-			return nil, err
-		}
-		hash = header.Hash()
-	}
 	// Construct the range filter
-	filter := b.r.filterSystem.NewBlockFilter(hash, addresses, topics)
+	filter := b.r.filterSystem.NewBlockFilter(b.hash, addresses, topics)
```

Comment:
And here you access `b.hash` directly again. What am I missing?

Reply:
You're not missing anything. I was missing some stuff :)